pandas/tests/io/test_parquet.py

```diff
@@ -1155,3 +1155,13 @@ def test_use_nullable_dtypes_not_supported(self, fp):
             df.to_parquet(path)
             with pytest.raises(ValueError, match="not supported for the fastparquet"):
                 read_parquet(path, engine="fastparquet", use_nullable_dtypes=True)
+
+    def test_close_file_handle_on_read_error(self):
+        df = pd.DataFrame({"a": [1, 2]})
+        with tm.ensure_clean() as path:
+            df.to_parquet(path)
+            with open(path, 'r+b') as f:
+                f.seek(16)
+                f.write(b'breakit')
+            with pytest.raises(Exception):  # Not important which exception
+                read_parquet(path, engine="fastparquet")
```

Comment:
Two very small comments/suggestions:

- Could use Path(path).write_bytes(b"breakit") (just my personal preference - feel free to ignore)
- Can you explicitly call Path(path).unlink() and add a comment that this would fail on windows if there is still an open file handle (just in case `ensure_clean` changes in the future and catches any exceptions when removing files).

Reply:
great points, done.